cdragontoolbox/binfile.py

```diff
@@ -131,6 +134,15 @@ def getv(self, key, default=None):
         except KeyError:
             return default
 
+    def getr(self, key, default=None):
```

Comment:
used for getting by-hash, since the normal get fails with KeyError if you use a hash

Reply:
`key_to_hash()` already handles strings, integers (hash values) and `BinHashBase` (wrapped bin values).
This means you can already use `get(0x12345678)` instead of `getr("12345678")` to access fields by hash (same for `getv()` vs. `getrv()`.

You should need to use hex strings for hash values, just use plain integers (as everywhere else) and you'll be fine. (Bonus: it will be faster.)

Also, you can probably use `item[key]` instead of `item.get(key)` most of the time.
The difference is that `[...]` will fail if the key is not found but `get()` will return a default value (`None` by default). If you know the key must exist, `[...]` is shorter.

Reply:
I'd like to keep use hashes since thats what is used elsewhere in cdragon

Reply:
You can use hashes, just as integers instead of strings. And that would just be in the script: no problem to output them as strings in the resulting JSON.